### PR TITLE
Added find_release_version_available function when version is release

### PR DESCRIPTION
**Pull request by ecornely**
##### SUMMARY
We need to download a release version without knowing exactly what it is so added the opportunity to use the keyword "release"

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
maven_artifact

##### ANSIBLE VERSION
```

```

### CODE REVIEW

**Comment by tumbl3w33d** [comment]
Hi, uhm, maven offers the keywords `LATEST` and `RELEASE` and I think I used that few weeks ago and it did exactly what you ask for. Try setting `RELEASE` as requested version, please. And get back to correct me if I'm wrong. :)

**Comment by ecornely** [comment]
We've first tried release than RELEASE but it didn't work and anyway by reading the python code of the module you can easily view the specific behavior when version == "latest".

The thing is that this module doesn't use any kind of maven btw it just mimic its way of resolving artifact

**Comment by tumbl3w33d** [comment]
Ha, you're right and while testing your branch I remembered why I was able to pull it off using `RELEASE`. I got it all wrapped in a `download_artifact` kind of role that does a lot of magic and I was probably targeting our old Nexus2 instance which still offers the good old API to which you can pass the maven-like version parameter. :)

Anyway, in general your implementation works. I'd like to hear the view of @turb on the expected argument `release` here. Shouldn't we try to mimic the maven-style keywords `LATEST` and `RELEASE` in addition to the existing `latest`? Possible implementation:
`latest` does what is does now - parsing the version nodes in the xml taking the newest.
`LATEST` does what the here proposed impl does, but looks at the dedicated `<latest>` node which can exist in the metadata. It should probably fail if the node doesn't exist. Alternatively it could silently fall back to `latest` behavior. I think I'd prefer failing.
`RELEASE` does what this PR proposes.

**Comment by ecornely** [comment]
Well if we have a lot of time to spend on that it would even be better to implement the complete version handling with version ranges (https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges) or like gradle as proposed in https://github.com/ansible/ansible/issues/20003

**Comment by tumbl3w33d** [comment]
Yeah for the sake of completeness that makes sense. I'd argue, that as long as there's no demand for that by anyone asking for it, we should postpone that to someday. It introduces complexity and might never be used by anyone. Or do you see the need for that?

Edit: oh, I just saw your last link points to such demand. :> We can still do that in a separate PR if someone really feels like adding it.

**Comment by turb** [comment]
tl;dr: Nexus and Artifactory may have different behaviour generating `latest` field in `maven-metadata.xml` depending on the scenario. I agree `latest` and `release` keywords are adequate, but it should be explicit in the documentation that it will use the corresponding field from the published `maven-metadata.xml` to avoid any confusion.

Long story:

`LATEST` and `RELEASE` have been dropped in Maven 3.

I had a look on `maven-metadata.xml` from our Artifactory virtual repo, and:
- the artifacts with only -SNAPSHOT versions have the `latest` field set, and no `release` field set.
- the artifacts with only release versions have both the `latest` and `release` field set to the same version.
- the artifacts with **both** release and -SNAPSHOT versions have both the `latest` and `release` field set **to the same release version, even if** a -SNAPSHOT is newer than the last release version.

It seems that with Artifactory the `latest` field already contains the latest release if one is available. [That's not what the Maven doc requires](https://maven.apache.org/ref/3.2.5/maven-repository-metadata/repository-metadata.html): we should have `latest` (incl. snapshots), `release` and `snapshot` fields.

I think it happens when Artifactory have separate release and snapshot repos: when provided a common virtual repo, it does not merge the two `maven-metadata.xml` but pick the release one.

**Comment by grossws** [comment]
@turb it seems Nexus 3 places correct `latest` and `release` versions:
```xml
<metadata modelVersion="1.1.0">
  <groupId>org.example</groupId>
  <artifactId>example-base</artifactId>
  <versioning>
    <latest>0.12.5-SNAPSHOT</latest>
    <release>0.12.4</release>
    <versions>
      <version>0.12.4</version>
      <version>0.12.5-SNAPSHOT</version>
    </versions>
    <lastUpdated>20190312175246</lastUpdated>
  </versioning>
</metadata>
```

In some cases I need to deploy latest release artifact, so my +1 for this PR, even if it's temporary remedy before maven ranges support. E. g. I deploy latest release version on prod and latest snapshot on staging, both through nexus.

**Comment by turb** [comment]
I am not confortable with the idea to add another non-standard keyword to be added temporarly, in order to be deprecated and removed one day.

If we add this, the documentation shoud be really bold about what it does and all this.
